CenterForOpenScience/ember-osf-web
# [ENG-4265] Duplicate subscription ids

-   Ticket: [4265]
-   Feature flag: n/a

## Purpose
- Prevent issues with duplicated subscription IDs for collection-providers and registration-providers

## Summary of Changes
- Create new models for provider-specific subscriptions
  - `registration-subscriptions` and `collection-subscriptions`
  - Update `<Subscriptions::Manager>` to handle these subscription types

## Screenshot(s)
- NA

## Side Effects
- None
## QA Notes
- Please verify that collection and registration moderators are able to update their notification frequencies
